src/cards/AbstractCard.ts

```diff
@@ -0,0 +1,62 @@
+import {Helper} from "../Helper";
+import {cards} from "../types/cards";
+import {generic} from "../types/generic";
+
+/**
+ * Abstract Card Class
+ *
+ * To create a new card, extend the new class with this one.
+ *
+ * @class
+ * @abstract
+ */
+class AbstractCard {
+    /**
+     * Entity to create the card for.
+     *
+     * @type {generic.hassEntity | generic.areaEntity}
+     */
+    entity: generic.hassEntity | generic.areaEntity;
```

Comment:
This combined type is duplicated four times. It could be an idea to extract it as a new type. Either in this file, or in generic.ts.
```
type Entity = generic.hassEntity | generic.areaEntity;
```

Reply:
The types (and code) still need to be optimized.

Currently, I'm investigating if I can use/port types from the Home Assistant [frontend Repository](https://github.com/home-assistant/frontend) instead of defining them in `generic`.

There's also the question if the @type declarations are still required for the code's self-documentation features. This will reduce the amount of duplicates also.

Thank you for reviewing!
I hope you're also testing it for functionality and errors.